docs/src/operate/c8y/supported-operations.md

```diff
@@ -241,9 +243,177 @@ Arguments will be parsed correctly as long as following features are not include
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command.template"
+[exec]
+topic = "c8y/devicecontrol/notifications"
+on_fragment = "c8y_Command"
+
+[exec.workflow]
+operation = "command"
+input = "${.payload.c8y_Command.text}"
+```
+
+To indicate that devices are supporting the operation, the device must publish a command capability message where command name is derived from `exec.workflow.operation`:
+
+```sh
+tedge mqtt pub -r 'te/device/<name>///cmd/command' '{}'
+```
+
+On receiving, mapper creates a symlink at `/etc/tedge/operations/c8y/c8y_Command` (for main device) or `/etc/tedge/operations/c8y/<external_id>/c8y_Command` (for child device) to master template file.
+
+The Mapper detects the change in the `/etc/tedge/operations/c8y` directory except for `.template` file. When the change is detected, `114` (supported operation) message is sent to c8y.
+
+After that, C8Y Json message is converted to thin-edge command, preserving all parameters provided in `input` field.
+
+```text title="Topic"
+te/device/<name>///cmd/command/c8y-mapper-1800380
+```
+
+```json5 title="Payload"
+{
+      "status": "init",
+      "text":"echo helloworld",
+}
+```
+
+:::info
+An operation file definition can contain multiple `input` fields:
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command.template"
+[exec]
+topic = "c8y/devicecontrol/notifications"
+on_fragment = "c8y_Command"
+
+[exec.workflow]
+operation = "command"
+input.x = "${.payload.c8y_Command.text}"
+input.y = { foo = "bar" }
+```
+
+Conversion to thin-edge command will contain both of them:
+
+```text title="Topic"
+te/device/<name>///cmd/command/c8y-mapper-1800380
+```
+
+```json5 title="Payload"
+{
+      "status": "init",
+      "text":"echo helloworld",
+      "foo":"bar",
+}
+```
+
+Make sure that provided input is json object, otherwise the operation execution will be skipped.
+:::
+
+After conversion, the `tedge-agent` handles custom operation commands using the workflow definition at `/etc/tedge/operations/command.toml`.
+Workflow definition must be created by the user.
+
+Here is a sample workflow for `C8y_Command`:
+
+```toml
+operation = "command"
+
+[init]
+action = "proceed"
+on_success = "executing"
+
+[executing]
+script = "/etc/tedge/operations/command ${.payload.c8y_Command.text}"
+on_success = "successful"
+
+[successful]
+action = "cleanup"
+
+[failed]
+action = "cleanup"
+```
+
+##### List of currently supported operations parameters
+Available in all C8Y Json operation files:
+* `topic` - The topic on which the operation will be executed (by default: `c8y/devicecontrol/notifications`).
+* `on_fragment` - Used to check if the mapping file matches the input payload.
+* `skip_status_update` - Optional boolean value that decide whether or not mapper should send operation status update messages (`501`-`503`/`504`-`506`).
+
+Available only in standard operation file definition:
```

Comment:
What is "standard operation file"?

Reply:
I am struggling to find a good name for that. I meant those JSON over MQTT operations which are not templates. I know standard is bad name for that but it supposed to be removed once I find better one.

Reply:
I changed it to `non-template operation file definition`.